### DISCUSSION TITLE
Import my nodes of ROS2

### DISCUSSION THREAD

**Discussion by oriolorra** [MOV-AI/movai-flow#221] [Q&A] [answered]
Hi,

I am new into mov.ai Flow. Seems a pretty good tool.

I have some questions:

- I am using ros2 humble, by default, ros2 has some packages, nodes,... But, I cannot see them inside Mov.ai Flow. Is it using only packages, nodes,... which are inside the movai flow docker?
- Is it using ROS1, or can be used with ROS2?
- If I have a workspace with some packages with nodes, launch files,.... How could I import these local packages?

Thanks

**Comment by MithunKinarullathil** [comment, marked as answer]
Hi @oriolorra, Thank you for the question and the appreciation.

- Currently, we are with ROS noetic, so you won't find any default ros2 nodes as the sources are not added by default. The container you are looking for is `spawner-robot1`, where we have all the executables from ros. Yes, the nodes you see in the IDE use the installed nodes inside the spawner container.
- At the moment by default, we are using ROS1, but we are planning to add ROS2 in the future, I can't tell you when at this point :(
- For importing the nodes you can follow [this procedure](https://flow.mov.ai/docs/importing-a-ros-node). The launch files are represented as flows in mov.ai, so you will need to make them using the IDE itself.

Let me know if you have further questions, I hope you find the product useful.
